File: packages/evals/tasks/bench/act/nonsense_action.ts
import { defineBenchTask } from "../../../framework/defineTask.js";

export default defineBenchTask(
  { name: "nonsense_action" },
  async ({ debugUrl, sessionUrl, stagehand, page, logger }) => {
    try {
      await page.goto("https://www.homedepot.com/");

      const { data: result } = await stagehand.act("what is the capital of the moon?");

      return {
        _success: !result.success, // We expect this to fail
        debugUrl,
        sessionUrl,
        logs: logger.getLogs(),
      };
    } catch (error) {
      return {
        _success: false,
        error: error instanceof Error ? error.message : String(error),
        debugUrl,
        sessionUrl,
        logs: logger.getLogs(),
      };
    } finally {
      await stagehand.close();
    }
  },
);

File: packages/sdk-ruby/evals/tasks/act/nonsense_action.rb
# frozen_string_literal: true

# Port of packages/evals/tasks/bench/act/nonsense_action.ts

Evals.define_task("nonsense_action") do |t|
  t.page.goto("https://www.homedepot.com/")

  result = t.stagehand.act("what is the capital of the moon?").data

  # We expect this to fail
  { _success: !result.success }
end
